options/options.go

```diff
@@ -302,8 +302,40 @@ func (terragruntOptions *TerragruntOptions) Clone(terragruntConfigPath string) *
 	}
 }
 
+// Check if argument is planfile TODO check file format
+func checkIfPlanfile(arg string) bool {
+	stat, err := os.Stat(arg)
+
+	if err != nil {
+		return false
+	}
+
+	return !stat.IsDir()
+}
+
+// Extract planfile from arguments list
+func extractPlanFile(argsToInsert []string) (*string, []string) {
+	planFile := ""
+	var filteredArgs []string
+
+	for _, arg := range argsToInsert {
+		if checkIfPlanfile(arg) {
+			planFile = arg
+		} else {
+			filteredArgs = append(filteredArgs, arg)
+		}
+	}
```

Comment:
Wouldn't this also extract out var files?

Reply:
@yorinasub17 , var files args are passed in `-var-file=filename` so `util.IsFile()` shouldn't return false

Reply:
Also var files can't be specified with planfile

Reply:
I've added tests which checks argument lists order

Reply:
You can pass in var files as `terragrunt -var-file filename` too, which I believe would fall into this case, right?

Reply:
No, it'll fail with `flag provided but not defined: -var-file` error. Anyway, I'll special check for tfvars files